test/integration/apiserver/flowcontrol/fight_test.go

```diff
@@ -0,0 +1,219 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flowcontrol
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	flowcontrol "k8s.io/api/flowcontrol/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/apimachinery/pkg/util/wait"
+	fcboot "k8s.io/apiserver/pkg/apis/flowcontrol/bootstrap"
+	genericfeatures "k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	utilfc "k8s.io/apiserver/pkg/util/flowcontrol"
+	fqtesting "k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/testing"
+	"k8s.io/apiserver/pkg/util/flowcontrol/metrics"
+	"k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	"k8s.io/klog/v2"
+)
+
+const timeFmt = "2006-01-02T15:04:05.999"
+
+func TestConfigConsumerFight(t *testing.T) {
+	// Disable the APF FeatureGate so that the normal config consumer
+	// controller does not interfere
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.APIPriorityAndFairness, false)()
+	_, loopbackConfig, closeFn := setup(t, 100, 100)
+	defer closeFn()
+	ctx := context.Background()
+	const size = 3
+	stopCh := make(chan struct{})
+	now := time.Now()
+	clk := clock.NewFakeClock(now)
+	ctlrs := map[bool][]utilfc.TestableInterface{
+		false: make([]utilfc.TestableInterface, size),
+		true:  make([]utilfc.TestableInterface, size)}
+	// maps FS key -> last written ResourceVersion
+	var lastRVs map[string]string = nil
+	// maps invert -> i -> FS key -> last notified ResourceVersion
+	notifiedRVs := map[bool][]map[string]string{
+		false: make([]map[string]string, size),
+		true:  make([]map[string]string, size)}
+	foreach := func(visit func(invert bool, i int)) {
+		for i := 0; i < size; i++ {
+			// The order of the following iteration is not deterministic,
+			// and that is good.
+			for invert := range ctlrs {
+				visit(invert, i)
+			}
+		}
+	}
+	foreach(func(invert bool, i int) {
+		myConfig := rest.CopyConfig(loopbackConfig)
+		myConfig = rest.AddUserAgent(myConfig, fmt.Sprintf("invert=%v, i=%d", invert, i))
+		myClientset := clientset.NewForConfigOrDie(myConfig)
+		fcIfc := myClientset.FlowcontrolV1beta1()
+		fsIfc := fcIfc.FlowSchemas()
+		if lastRVs == nil {
+			lastRVs = make(map[string]string)
+			// Wait until every FlowSchema is defined, and record its RV
+			allFlowSchemas := append(fcboot.MandatoryFlowSchemas, fcboot.SuggestedFlowSchemas...)
+			err := wait.Poll(time.Second, wait.ForeverTestTimeout, func() (bool, error) {
+				for _, fs := range allFlowSchemas {
+					_, err := fsIfc.Get(ctx, fs.Name, metav1.GetOptions{})
+					if err != nil {
+						return false, err
+					}
+					t.Logf("Got initial FlowSchema %#+v", fs)
```

Comment:
This log line is printing your stub resource instance, not the resource instance that you got on line 87, which is why metadata is empty.

Reply:
fixed